src/wordlist.c

```diff
@@ -550,6 +552,23 @@ void do_wordlist_crack(struct db_main *db, const char *name, int rules)
 	}
 
 	if (name) {
+		struct stat st;
+
+		if (!(word_file = jtr_fopen(path_expand(name), "rb")))
+			pexit(STR_MACRO(jtr_fopen)": %s", path_expand(name));
+
+		if (fstat(fileno(word_file), &st))
+			pexit("fstat");
+
+		file_is_fifo = ((st.st_mode & S_IFMT) == S_IFIFO);
```

Comment:
That's still not it - we need `file_is_fifo` to be reset to 0 if `!name` as well. So just set it to 0 before the `if (name)`.

Reply:
Oh, right

Reply:
There'll likely be lots of other things going wrong in wordlist that day though 🤣